compose/foundation/foundation/src/commonMain/kotlin/androidx/compose/foundation/gestures/MouseWheelScrollable.kt

```diff
@@ -130,36 +133,28 @@ private class RawMouseWheelScrollPhysics(
 }
 
 private class AnimatedMouseWheelScrollPhysics(
-    override var mouseWheelScrollConfig: ScrollConfig,
-    override var scrollingLogic: ScrollingLogic,
+    private val coroutineScope: CoroutineScope,
+    mouseWheelScrollConfig: ScrollConfig,
+    scrollingLogic: ScrollingLogic,
     val density: () -> Density,
-) : ScrollPhysics() {
-    private var isAnimationRunning = false
+) : ScrollPhysics(mouseWheelScrollConfig, scrollingLogic) {
     private val channel = Channel<Float>(capacity = Channel.UNLIMITED)
 
-    override suspend fun launch() {
+    fun launchAnimatedDispatchScroll() = coroutineScope.launch {
         while (coroutineContext.isActive) {
-            val event = channel.receive()
-            isAnimationRunning = true
-            try {
-                scrollingLogic.animatedDispatchScroll(event, speed = 1f * density().density) {
-                    // Sum delta from all pending events to avoid multiple animation restarts.
-                    channel.sumOrNull()
-                }
-            } finally {
-                isAnimationRunning = false
+            val eventDelta = channel.receive()
+            scrollingLogic.animatedDispatchScroll(eventDelta, speed = 1f * density().density) {
+                // Sum delta from all pending events to avoid multiple animation restarts.
+                channel.sumOrNull()
             }
         }
     }
 
-    override fun PointerInputScope.onMouseWheel(pointerEvent: PointerEvent): Boolean {
-        val scrollDelta = with(mouseWheelScrollConfig) {
-            calculateMouseWheelScroll(pointerEvent, size)
-        }
-        return if (mouseWheelScrollConfig.isPreciseWheelScroll(pointerEvent)) {
-            // In case of high-resolution wheel, such as a freely rotating wheel with no notches
-            // or trackpads, delta should apply directly without any delays.
-            scrollingLogic.dispatchRawDelta(scrollDelta) != Offset.Zero
+    private fun ScrollingLogic.dispatchPreciseWheelScroll(scrollDelta: Offset) {
+        coroutineScope.launch {
```

Comment:
This most probably causes input lags and junks on macOs:

https://github.com/JetBrains/compose-multiplatform-core/assets/5963351/b4c1a2ae-5411-46c1-9e71-a33e180101a5

(from https://github.com/JetBrains/compose-multiplatform-core/tree/ivan.matkov/fix-nested-scroll-compare-with-old).

cc @m-sasha let's discuss tomorrow a possible fix. We should flush coroutines not only from composition, but from events as well.

Until then, we can't merge the PR. After fixing, we can merge the PR as it is if the junks are fixed.

Reply:
I investigated if this helps to remove junks:
```
Subject: [PATCH] -
---
Index: compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/scene/BaseComposeScene.skiko.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/scene/BaseComposeScene.skiko.kt b/compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/scene/BaseComposeScene.skiko.kt
--- a/compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/scene/BaseComposeScene.skiko.kt	(revision da04be1b94c73ebace27a96766f3b84048342e09)
+++ b/compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/scene/BaseComposeScene.skiko.kt	(date 1707517409453)
@@ -159,6 +159,7 @@
         // dispatched, in order to allow registering for these events before they are sent.
         // Otherwise, events like a synthetic mouse-enter sent due to a new element appearing under
         // the pointer would be missed by e.g. InteractionSource.collectHoverAsState
+        recomposer.performScheduledEffects()     // Composition effects (e.g. LaunchedEffect)
         recomposer.performScheduledTasks()
         frameClock.sendFrame(nanoTime)           // Recomposition
         doLayout()                               // Layout
```
It doesn't help unfortunately. I need to compare timings of events and frames to remove subjectivity, and understand the reason.